plugins/rts54hub/fu-rts54hub-rtd21xx-device.c

```diff
@@ -541,6 +874,22 @@ fu_rts54hub_rtd21xx_device_write_firmware (FuDevice *device,
 	return TRUE;
 }
 
+
+static gboolean
+fu_rts54hub_rtd21xx_device_write_firmware (FuDevice *device,
+					   FuFirmware *firmware,
+					   FwupdInstallFlags flags,
+					   GError **error)
+{
+
+	FuRts54hubRtd21xxDevice *self = FU_RTS54HUB_RTD21XX_DEVICE (device);
+	if(self->fwupdate_mode == FW_UPDATE_BACKGROUND_MODE)
+		return fu_rts54hub_rtd21xx_device_write_firmware_background(device, firmware, flags, error);
+	else
+		return fu_rts54hub_rtd21xx_device_write_firmware_foreground(device, firmware, flags, error);
+	return TRUE;
+}
```

Comment:
this else block seems redundant.  Might I suggest a different way?


```
return (self->fwupdate_mode == FW_UPDATE_BACKGROUND_MODE) ? fu_rts54hub_rtd21xx_device_write_firmware_background (..) : fu_rts54hub_rtd21xx_device_write_firmware_foreground (...);
```

That will be more readable

Reply:
OK